src/service/mainQueryService.js

```diff
@@ -0,0 +1,55 @@
+import {Product} from "../class/Product"
+import _ from "lodash";
+import {getLinksByAccountIdsWithProduct} from "../db/dataService/mainQueryDataService";
+
+export async function createUsersSearchs(){
+    const resultDB=await getLinksByAccountIdsWithProduct();
+    return _.groupBy(resultDB,'link')
+}
+
+export function createSearchInformation(resultDB){
+
+    const pageQuantity=createPageQuantityFromResultDB(resultDB)
+    const productFromDb=parsProductFromDB(resultDB);
+    const idAccount=parsIdAccount(resultDB);
+    const idSearch=parsIdSearch(resultDB);
+    const email=parsEmail(resultDB);
```

Comment:
Tu jest na pewno coś nie tak, ze w kazdej z tych funkcji robisz groupBy, wrzuć do komentarza przykładowe `resultDb` to zobaczę o co tam chodzi